packages/vite/src/node/plugins/css.ts

```diff
@@ -1178,11 +1169,11 @@ async function resolvePostcssConfig(
           throw new Error(`Failed to load PostCSS config: ${e}`)
         }
       }
-      result = null
-    }
+      return null
+    })
   }
 
-  postcssConfigCache[dialect].set(config, result)
+  postcssConfigCache.set(config, result)
```

Comment:
Maybe we could set the cache to the result to avoid keeping the Promise around. It is better to avoid caching promises long term.
```suggestion
  postcssConfigCache.set(config, result)
  result.then((resolved) => {
    postcssConfigCache.set(config, resolved)
  })
```

Reply:
Good idea 👍